### PR TITLE
Sets warnings as errors and adds FxCop analyzer to Schema project.

### PR DESCRIPTION
**Pull request by gabog**
Relates to #2367.

- Configured TreatWarningsAsErrors on all projects in the libraries folder except for Microsoft.Bot.Builder.Dialogs.Adaptive.Testing (that needs to be addressed separately).
- Created Directory.Build.props for FunctionalTests and Tests to manage excludes for the projects in those folders.
- Edited projects to generate XML documentation for debug config too so we can catch this early on rather than later in the build server. 
- Added NoWarn properties for documentation on each project (these will be addressed by #4052).
- Fixed long hanging fruit warnings.
- Added Roslyn FxCop analyzer to Microsoft.Bot.Schema and fixed violations or excluded them using #pragma with justification for the exclude (most of the changes couldn't be made because of binary compat).

### CODE REVIEW

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Webex/Microsoft.Bot.Builder.Adapters.Webex.csproj [hunk 1]
```diff
@@ -6,28 +6,28 @@
...
-    <WarningsAsErrors />
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
     <DebugType>Full</DebugType>
     <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <!-- CS8002: The Thrzn41.WebexTeams package isn't signed, so supress the warning. There seems to not be a way to supress this for ONLY Thrzn41.WebexTeams. -->
```
Comment (carlosscastro) [inline libraries/Adapters/Microsoft.Bot.Builder.Adapters.Webex/Microsoft.Bot.Builder.Adapters.Webex.csproj:26]: Why don't people sign their software? This and slack are pretty surprising to see unsigned.

File: libraries/Microsoft.Bot.Builder.ApplicationInsights/Microsoft.Bot.Builder.ApplicationInsights.csproj [hunk 2]
```diff
@@ -7,6 +7,8 @@
     <PackageVersion Condition=" '$(IsBuildServer)' != '' ">$(ReleasePackageVersion)</PackageVersion>
     <Configurations>Debug;Release</Configurations>
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+    <DocumentationFile>bin\$(Configuration)\netstandard2.0\Microsoft.Bot.Builder.ApplicationInsights.xml</DocumentationFile>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
   </PropertyGroup>
 
   <PropertyGroup>
```
Comment (mrivera-ms) [inline libraries/Microsoft.Bot.Builder.ApplicationInsights/Microsoft.Bot.Builder.ApplicationInsights.csproj:14]: Do we need to suppress documentation warnings for this project?

Comment (gabog) [reply libraries/Microsoft.Bot.Builder.ApplicationInsights/Microsoft.Bot.Builder.ApplicationInsights.csproj:14]: No, all of the documentation is there, any new changes that don't have docs will be treated as errors going forward, makes sense?

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Webex/Microsoft.Bot.Builder.Adapters.Webex.csproj (hunk 1 above)
Comment (garypretty) [reply libraries/Adapters/Microsoft.Bot.Builder.Adapters.Webex/Microsoft.Bot.Builder.Adapters.Webex.csproj:26]: Its a 3rd party package - many 3rd party packages are not signed - sadly there is no good official client for webex (or slack / twilio) for that matter, so we hit this issue in all of the adapters.

File: libraries/Microsoft.Bot.Builder.ApplicationInsights/Microsoft.Bot.Builder.ApplicationInsights.csproj (hunk 2 above)
Comment (mrivera-ms) [reply libraries/Microsoft.Bot.Builder.ApplicationInsights/Microsoft.Bot.Builder.ApplicationInsights.csproj:14]: Perfect sense. Thanks.

---
In reply to: [439581296](https://github.com/microsoft/botbuilder-dotnet/pull/4061#discussion_r439581296) [](ancestors = 439581296)

**Comment by cleemullins** [comment]
Merging.

**Pull request merged**